### PR TITLE
Robogames

### LINKED ISSUE #81: Get dev up to date

### PR DESCRIPTION
**Pull request by blhough**
### all this code was added within 24 hours of the competition, so it got a little messy

This code, bar some clean up, is what ran at Robogames

closes #81
- [X] Compiles on desktop
- [X] Complies on Teensy
#### Additions
- Add Timer class
- moveForward with PID implementation
- rotate with PID implementation
- Add test functions for moveforward and rotate
- Max integral error for PID controller
- 4x IR calibration loading from EEPROM
#### Changes
- Fix the negative modulo bug
- Mapping uses random directions
- Fix isClearLeft/Right
- isWallInDirection no longer samples multiple times
- Correct whitespace

### CODE REVIEW

File: micromouse/Vector.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Micromouse
 
 	direction operator-(const direction &dir1, const direction &dir2)
 	{
-		return static_cast<direction>(((int)dir1 - (int)dir2) % 8);
+		return static_cast<direction>((64 + (int)dir1 - (int)dir2) % 8);
```
Comment (blhough) [inline micromouse/Vector.cpp:15]: this motherfucker

**Comment by Pastyguy** [comment]
What the hell does it do?

**Comment by blhough** [comment]
what?

File: micromouse/Vector.cpp (hunk 1 above)
Comment (blhough) [inline micromouse/Vector.cpp:15]: @Pastyguy : "What the hell does it do?"

it was the bug that cost us 3 hours

File: micromouse/RobotIO.cpp [hunk 2]
```diff
@@ -52,14 +57,14 @@ namespace Micromouse
 
 	bool RobotIO::isClearRight()
 	{
-		return !isWallinDirection(W);
+		return !isWallinDirection(E);
```
Comment (blhough) [inline micromouse/RobotIO.cpp:60]: Dammit  Seaney!

**Pull request merged**